api/v2beta1/helmrelease_types.go

```diff
@@ -824,15 +824,15 @@ func (in *HelmRelease) SetConditions(conditions []metav1.Condition) {
 
 // HelmReleaseAttempted registers an attempt of the given HelmRelease with the given state.
 // and returns the modified HelmRelease and a boolean indicating a state change.
-func HelmReleaseAttempted(hr HelmRelease, revision string, releaseRevision int, valuesChecksum string) (HelmRelease, bool) {
-	changed := hr.Status.LastAttemptedRevision != revision ||
-		hr.Status.LastReleaseRevision != releaseRevision ||
-		hr.Status.LastAttemptedValuesChecksum != valuesChecksum
-	hr.Status.LastAttemptedRevision = revision
-	hr.Status.LastReleaseRevision = releaseRevision
-	hr.Status.LastAttemptedValuesChecksum = valuesChecksum
+func HelmReleaseAttempted(obj *HelmRelease, revision string, releaseRevision int, valuesChecksum string) bool {
```

Comment:
I would prefer this to be removed from the API package in the end, as it is more related to the logic of the reconciler.